### DISCUSSION TITLE
How to handle pseudo-classes classes using Dripsy/Moti Link for web

### DISCUSSION THREAD

**Discussion by adherb** [nandorojo/solito#132] [Q&A]
What is the best option to handle the hover selector, I basically would like to change the background color on hover on web.

Should I be using Dripsy for this? Or should I be using Moti Link?

Is there an example of how I can implement this?

**Comment by nandorojo** [comment]
check out Moti Interactions! https://moti.fyi/interactions/overview

**Reply by nandorojo** [reply]
`MotiLink` is a wrapper around `MotiPressable`. If you want an animation, use one of those. If you just want basic hover styles for Web, you can also use `Pressable` directly from React Native.

**Reply by nandorojo** [reply]
I was on my phone earlier, but now I can give a few additional answers...

For Dripsy usage, you'll want to rely on `useDripsyTheme` and `useResponsiveValue`, since you can't use `useSx` inside of `MotiPressable`'s `animate` prop due to Reanimated worklet constraints.

**Reply by nandorojo** [reply]
As for animating background color, my typical method is to absolute position a second view with the different background color and then animate it in with opacity.

```tsx
import { useMotiPressable, MotiPressable } from 'moti/interactions'
import { MotiView } from 'moti'
import { useSx } from 'dripsy'
import { StyleSheet } from 'react-native'

const Background = () => {
  const sx = useSx()
  const state = useMotiPressable(({ hovered, pressed }) => {
    'worklet'

    return { opacity: hovered || pressed ? 1 : 0 }
  }, [])

  return (
     <MotiView 
       style={[sx({ bg: '$highlight', zIndex: -1 }), StyleSheet.absoluteFill]} 
       state={state} 
     />
  )
}

export const Element = () => {
  return (
    <MotiPressable>
      <Background />

     {/* ...the rest here */}
   </MotiPressable>
  )
}
```

**Reply by nandorojo** [reply]
There is a Dripsy section on the Moti docs: https://moti.fyi/interactions/merge#usage-with-dripsy
